### ISSUE TITLE
sorting appears broken with text based searching

### ISSUE THREAD

**Issue by Calemy** [meilisearch/meilisearch#4862]
**Describe the bug**
when sorting while performing a text based search; the return hits seem to be at random.

**Additional context**
to fill a test database with data feel free to pull the necessary data via ```https://catboy.best/api/v2/search?q=harumachi clover&limit=1000```

Issue has originally been described here:
https://github.com/meilisearch/meilisearch-js/issues/1687

**Comment by irevoire** [comment]
Hey @Calemy, thanks for the report.

Could you share your [ranking-rules](https://www.meilisearch.com/docs/reference/api/settings#ranking-rules) with us?

**Comment by Calemy** [comment]
using the default ranking rule as specified on your documentation.

**Comment by Calemy** [comment]
Hello @irevoire

I'm using the default ranking rule as specified on your documentation.

**Comment by irevoire** [comment]
Hey,

Here, you can see that with the default ranking rules, I also reproduced your issue:
<img width="1546" alt="image" src="https://github.com/user-attachments/assets/53697e17-d4c0-499c-87e6-166eedb722f7">

But, **this behavior is expected**.
As you can see in the default ranking-rule order:
<img width="192" alt="image" src="https://github.com/user-attachments/assets/a2092db3-3bbb-4146-ae88-431df3599096">
Meilisearch first sorts the documents by:
1. [Words](https://www.meilisearch.com/docs/learn/relevancy/ranking_rules#1-words) => how many words of the query does the document contain
2. [Typo](https://www.meilisearch.com/docs/learn/relevancy/ranking_rules#2-typo) => the documents with the least typos appear first
3. [Proximity](https://www.meilisearch.com/docs/learn/relevancy/ranking_rules#3-proximity)
4. [Attribute](https://www.meilisearch.com/docs/learn/relevancy/ranking_rules#4-attribute)
5. And only after all of that, [Sort](https://www.meilisearch.com/docs/learn/relevancy/ranking_rules#5-sort)

If you want your documents to first be sorted by `play_count` and then be sorted by relevancy, the immediate fix is to [re-order your ranking rule](https://www.meilisearch.com/docs/reference/api/settings#update-ranking-rules) so the first one to apply is sort: `{ "rankingRules": ["sort", "words", "typo", "proximity", "attribute", "exactness"] }`

After doing that, as you can see, my documents are sorted:
<img width="1558" alt="image" src="https://github.com/user-attachments/assets/8f85d4a8-87fd-469e-ae7c-96fceb809d71">


BUT you need to understand that by sorting the documents by a number, meilisearch will only be able to re-order two documents that share the exact same `play_count`. On my screenshot, for example, all the documents have different scores, which means my query is pretty much ignored. The right set of documents is retrieved, but then they're not sorted by relevancy because the sort rule doesn't allow the space for any other rules to apply.
In your case, a good idea is generally to [bucket or bin](https://en.wikipedia.org/wiki/Data_binning) your view_count.
Dividing your view count by `1000`, for example, before inserting your documents into meilisearch, would let meilisearch rerank more documents.

Anyway, I believe your issue is fixed; happy relevancy tweaking 🎉 
And feel free to ask more questions about the issue tweaking your relevancy. Or open a new issue if you believe you found another bug
